src/HighTable.tsx

```diff
@@ -144,7 +105,14 @@ export default function HighTable({
    * - data.rows(dataIndex, dataIndex + 1)
    * - data.rows(tableIndex, tableIndex + 1, orderBy)
    */
-  const { slice, invalidate, hasCompleteRow, data: previousData } = state
+
+  // true if at least one row is fully resolved (all of its cells)
```

Comment:
I think this is technically running more often than it needs to. It only needs to run once-per-dataframe so that the TableHeader knows when to measure the widths. Right now it's running everytime the slice changes. (I think this was true before this PR too, but didn't require an extra iteration). Not saying to change it now, just something to think about.

Reply:
Indeed! I thought the same but had the feeling that it was a premature optimization. I can look at it again though

Reply:
OK, fixed!